src/marlin_stubs/pause/M600.cpp

```diff
@@ -116,10 +119,48 @@ void GcodeSuite::M600() {
 }
 
 void M600_execute(xyz_pos_t park_point, int8_t target_extruder,
-    xyze_float_t resume_point, std::optional<float> unloadLength, std::optional<float> fastLoadLength, std::optional<float> retractLength,
-    pause::Settings::CalledFrom);
+    xyze_float_t resume_point, std::optional<float> unloadLength, std::optional<float> fastLoadLength,
+    std::optional<float> retractLength, std::optional<filament::Colour> filament_colour,
+    std::optional<filament::Type> filament_type, pause::Settings::CalledFrom);
 
 void M600_manual() {
+    char filamenttype[16] = { '\0' };
+    char colourtype[16] = { '\0' };
+
+    auto filament_to_be_loaded = filament::Type::NONE;
+    const char *text_begin = 0;
+    if (parser.seen('S')) {
+        text_begin = strchr(parser.string_arg, '"');
+        if (text_begin) {
+            ++text_begin; // move pointer from '"' to first letter
+            const char *text_end = strchr(text_begin, '"');
+            if (text_end) {
+                auto filament = filament::get_type(text_begin, text_end - text_begin);
+                if (filament != filament::Type::NONE) {
```

Comment:
Is this particular `if` needed for anything?

Reply:
Yes, it is needed. It does make sure that the filament type was used correctly.
`M600 S"PLA"` is fine. However if `M600 S"PLA` is sent without quote for whatever reason, it shouldn't continue.

Reply:
My point here is ‒ if the result is `Type::NONE` and the `filament_to_be_loaded` is initialized to `Type::NONE`, if the condition wasn't there, it'd simply overwrite the old `NONE` with a new `NONE` 😇.

But that doesn't matter.